src/osx/cocoa/textctrl.mm

```diff
@@ -1210,6 +1234,26 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
                 break;
         }
     }
+    if( style.HasLeftIndent() )
+    {
+        if( start == end && start == 0 )
+            range = NSMakeRange( start, start + 1 );
+        else if( start == end && start == m_textView.textStorage.string.length )
+            range = NSMakeRange( start - 1, 1 );
+        else
+            range = [[m_textView textStorage].string paragraphRangeForRange: NSMakeRange( start, 1 )];
+        paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+        [paragraphStyle setFirstLineHeadIndent: indent];
+        [paragraphStyle setHeadIndent: indent];
+        [storage addAttribute: NSParagraphStyleAttributeName value: paragraphStyle range: range];
+        if( /*start == -1 && end == -1*/range.length == 0 )
```

Comment:
What does this commented out code mean? What are you actually doing here? How can range length be 0 when you always create it with `len = 1` above?